src/DisplayFormats/DisplayFormats.cpp

```diff
@@ -22,27 +24,77 @@ std::string GetDisplaySize(uint64_t size_bytes) {
   return absl::StrFormat("%.2f TB", size_bytes / kBytesInTb);
 }
 
-std::string GetDisplayTime(absl::Duration duration) {
+constexpr double kHoursInOneDay = 24;
+
+TimeUnit ChooseUnitForDisplayTime(absl::Duration duration) {
   if (duration < absl::Microseconds(1)) {
-    return absl::StrFormat("%.3f ns", absl::ToDoubleNanoseconds(duration));
+    return TimeUnit::kNanosecond;
   }
   if (duration < absl::Milliseconds(1)) {
-    return absl::StrFormat("%.3f us", absl::ToDoubleMicroseconds(duration));
+    return TimeUnit::kMicrosecod;
   }
   if (duration < absl::Seconds(1)) {
-    return absl::StrFormat("%.3f ms", absl::ToDoubleMilliseconds(duration));
+    return TimeUnit::kMillisecond;
   }
   if (duration < absl::Minutes(1)) {
-    return absl::StrFormat("%.3f s", absl::ToDoubleSeconds(duration));
+    return TimeUnit::kSecond;
   }
   if (duration < absl::Hours(1)) {
-    return absl::StrFormat("%.3f min", absl::ToDoubleMinutes(duration));
+    return TimeUnit::kMinute;
   }
-  constexpr double kHoursInOneDay = 24;
   if (duration < absl::Hours(kHoursInOneDay)) {
-    return absl::StrFormat("%.3f h", absl::ToDoubleHours(duration));
+    return TimeUnit::kHour;
+  }
+  return TimeUnit::kDay;
+}
+
+std::string DisplayTimeUnit(TimeUnit unit) {
+  switch (unit) {
+    case TimeUnit::kNanosecond:
+      return "ns";
+    case TimeUnit::kMicrosecod:
+      return "us";
+    case TimeUnit::kMillisecond:
+      return "ms";
+    case TimeUnit::kSecond:
+      return "s";
+    case TimeUnit::kMinute:
+      return "min";
+    case TimeUnit::kHour:
+      return "h";
+    case TimeUnit::kDay:
+      return "days";
+    default:
+      ORBIT_UNREACHABLE();
   }
-  return absl::StrFormat("%.3f days", absl::ToDoubleHours(duration) / kHoursInOneDay);
+}
+
+double ToDoubleTimeUnits(absl::Duration duration, TimeUnit unit) {
+  switch (unit) {
+    case TimeUnit::kNanosecond:
+      return absl::ToDoubleNanoseconds(duration);
+    case TimeUnit::kMicrosecod:
+      return absl::ToDoubleMicroseconds(duration);
+    case TimeUnit::kMillisecond:
+      return absl::ToDoubleMilliseconds(duration);
+    case TimeUnit::kSecond:
+      return absl::ToDoubleSeconds(duration);
+    case TimeUnit::kMinute:
+      return absl::ToDoubleMinutes(duration);
+    case TimeUnit::kHour:
+      return absl::ToDoubleHours(duration);
+    case TimeUnit::kDay:
+      return absl::ToDoubleHours(duration) / kHoursInOneDay;
+    default:
+      ORBIT_UNREACHABLE();
+  }
+}
+
+std::string GetDisplayTime(absl::Duration duration) {
+  const TimeUnit unit = ChooseUnitForDisplayTime(duration);
+  const std::string unit_str = DisplayTimeUnit(unit);
+  const double duration_double = ToDoubleTimeUnits(duration, unit);
```

Comment:
Nit/Personal preference: I'm not sure if having these two temporary variables makes it more readable.

Reply:
Fair enough.